docs/architecture/psa-migration/outcome-analysis.sh

```diff
@@ -43,8 +43,15 @@ record() {
     fi
 }
 
-# save current HEAD
-HEAD=$(git branch --show-current)
+# save current HEAD.
+# Note: unfortunately "git branch --show-current" was added only in GIT
+# version 2.22.
+GIT_VERSION="$(git --version | sed 's/git version //')"
+if dpkg --compare-versions "$GIT_VERSION" "gt" "2.22.0"; then
+    HEAD=$(git branch --show-current)
+else
+    HEAD=$(git rev-parse --abbrev-ref HEAD)
```

Comment:
Why not always use this command? Is there a case where `git branch --show-current` is better in some non-critical way?

We're planning to move most of our CI from Ubuntu 16.04 to 22.04 in the next quarter. We can move outcome analysis sooner if there's a strong benefit.

Reply:
I think it makes for slightly more understandable code if you can see that `git rev-parse --abbrev-ref HEAD` is equivalent to `git branch --show-current` for older versions of git. I did not personally know that before reviewing this PR for example. I think it is ok to leave it as is.

Reply:
A comment explaining what the command do would serve the same purpose and would reduce the complexity of the code.

The version check has added a dependency on `dpkg`, so it will fail for developers who work on non-Debian-based systems (e.g. Fedora, macOS, ...). Also, it's wrong if someone has installed a more recent `git` in their `PATH`.

Reply:
> A comment explaining what the command do would serve the same purpose and would reduce the complexity of the code.
> 
> The version check has added a dependency on `dpkg`, so it will fail for developers who work on non-Debian-based systems (e.g. Fedora, macOS, ...). Also, it's wrong if someone has installed a more recent `git` in their `PATH`.

Yep, fair enough on the `dpkg` part, I agree.